SU-SWS/ace-sdssgryphon
# SDSS-1299: Add Explore More options and display to News

# READY FOR REVIEW

# Summary
- Created curated and random views for Explore More so that content authors can display more relevant content and so that more content is surfaced to readers of news
- Created new field to allow author to choose which view to display, with random view as default

# Review By (Date)
- In time to release pre-Aug 31

# Criticality
-HIGH

# Review Tasks

## Setup tasks and/or behavior to test

### Test the random default view
1. Check out this branch
2. Create a bunch of news. Tag them with a variety of News Types
3. Verify that random/different news items show up in the Explore More section for each news item (not the same ones on every page)

### Test the contextual filter on the Random display
1. Edit one of the news items. 
2. Go to the Options for Explore More field
3. Enter an argument for one of the terms you used when you tagged news
4. Verify that the randomized view only contains news tagged with that term

### Test curated displays
1. Edit one of the news items
2. Under the Related Content field, select three other news items
3. Go to the Options for Explore More field, and select the Display "News that is selected as related on this article"
4. Verify that Explore More shows your selected news items
5. Now edit one of the news items that you selected as Related Content
3. Go to the Options for Explore More field, and select the Display "News that has selected this article as related"
4. Verify that Explire More shows only the news item from Step 1.


## Front End Validation
No new front-end code

## Backend / Functional Validation
### Code
- [ YES] Are the naming conventions following our standards?
- [ NOPE] Does the code have sufficient inline comments?
- [ NOPE] Is there anything in this code that would be hidden or hard to discover through the UI?
- [NOPE ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ NEED TESTS] Are tests provided? eg (unit, behat, or codeception)


## General
- [NOPE ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [SHRUG ] Is the approach to the problem appropriate?

# Affected Projects or Products
- SDSS News
- News throughout the SDSS stack

# Associated Issues and/or People
[- SDSS-1299](https://stanfordits.atlassian.net/browse/SDSS-1299)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)
